File: backend/app/schemas/contract.py
"""合同相关 schema。"""

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel


class ContractCreate(BaseModel):
    """Agent 推送合同提取结果。"""

    contract_no: str
    contract_type: str
    type_judge_basis: str | None = None
    party_a: str | None = None
    party_b: str | None = None
    our_role: str | None = None
    signing_date: date | None = None
    contract_period: str | None = None
    total_amount: float | None = None
    amount_uppercase: str | None = None
    tax_rate: float | None = None
    payment_terms: str | None = None
    delivery_terms: str | None = None
    acceptance_terms: str | None = None
    breach_terms: str | None = None
    warranty_terms: str | None = None
    ip_terms: str | None = None
    other_key_terms: str | None = None
    source_file_name: str | None = None


class ContractUpdate(BaseModel):
    """人工修正合同字段。所有字段可选。"""

    party_a: str | None = None
    party_b: str | None = None
    our_role: str | None = None
    signing_date: date | None = None
    contract_period: str | None = None
    total_amount: float | None = None
    amount_uppercase: str | None = None
    tax_rate: float | None = None
    payment_terms: str | None = None
    delivery_terms: str | None = None
    acceptance_terms: str | None = None
    breach_terms: str | None = None
    warranty_terms: str | None = None
    ip_terms: str | None = None
    other_key_terms: str | None = None


class ContractListItem(BaseModel):
    id: UUID
    contract_no: str
    contract_type: str
    party_a: str | None = None
    party_b: str | None = None
    total_amount: float | None = None
    city: str | None = None
    project_name: str | None = None
    verified: bool = False
    verified_by: str | None = None
    verified_at: datetime | None = None

    model_config = {"from_attributes": True}


class ContractDetail(ContractListItem):
    type_judge_basis: str | None = None
    our_role: str | None = None
    signing_date: date | None = None
    contract_period: str | None = None
    amount_uppercase: str | None = None
    tax_rate: float | None = None
    payment_terms: str | None = None
    delivery_terms: str | None = None
    acceptance_terms: str | None = None
    breach_terms: str | None = None
    warranty_terms: str | None = None
    ip_terms: str | None = None
    other_key_terms: str | None = None
    source_file_name: str | None = None
    extracted_at: datetime | None = None


class VerifyRequest(BaseModel):
    verified_by: str | None = None


class ProjectCardItem(BaseModel):
    """项目卡片数据：前后项合同并排展示。"""
    contract_no: str
    project_name: str | None = None
    city: str | None = None
    has_front_contract: bool = False
    has_back_contract: bool = False
    has_front_acceptance: bool = False
    has_back_acceptance: bool = False
    llm_analyzed: bool = False
    project_risk: str | None = None
    audit_status: str | None = None
    front_contract: ContractListItem | None = None
    back_contract: ContractListItem | None = None

File: frontend/src/types/contract.ts
/** 合同相关类型 */

export interface Contract {
  id: string
  contract_no: string
  contract_type: string
  party_a?: string
  party_b?: string
  total_amount?: number
  city?: string
  project_name?: string
  verified: boolean
  verified_by?: string
  verified_at?: string
}

export interface ContractDetail extends Contract {
  type_judge_basis?: string
  our_role?: string
  signing_date?: string
  contract_period?: string
  amount_uppercase?: string
  tax_rate?: number
  payment_terms?: string
  delivery_terms?: string
  acceptance_terms?: string
  breach_terms?: string
  warranty_terms?: string
  ip_terms?: string
  other_key_terms?: string
  source_file_name?: string
  extracted_at?: string
}

/** 项目卡片数据：前后项并排展示 */
export interface ProjectCard {
  contract_no: string
  project_name?: string
  city?: string
  has_front_contract: boolean
  has_back_contract: boolean
  has_front_acceptance: boolean
  has_back_acceptance: boolean
  llm_analyzed: boolean
  project_risk?: string
  audit_status?: string
  front_contract: Contract | null
  back_contract: Contract | null
}
